web/views.py

```
@@ -4681,55 +4690,78 @@ def donation_webhook(request):
+        elif event.type == "invoice.payment_failed":
+            invoice = event.data.object
+            logger.info(f"Processing failed invoice payment: {invoice.id}")
+            handle_invoice_failed(invoice)
+
+        return HttpResponse(status=200)
+    except Exception as e:
+        logger.error(f"Error processing webhook {event.type}: {str(e)}")
+        return HttpResponse(status=400)
 
 
 def handle_successful_donation_payment(payment_intent):
```

Ruff (0.8.2)

Missing return type annotation for public function `handle_successful_donation_payment`

Add return type annotation: `None`

(ANN201)